caikit/runtime/model_management/model_manager.py

```diff
@@ -114,8 +114,13 @@ def __init__(self):
         self._lazy_load_local_models = runtime_cfg.lazy_load_local_models
         error.value_check(
             "<RUN44773514E>",
+            # _lazy_load_local_models is set, and _local_models_dir is empty
             not self._lazy_load_local_models or self._local_models_dir,
-            "Must set runtime.local_models_dir with runtime.lazy_load_local_models",
+            (
+                "runtime.local_models_dir must be a valid path"
+                " if set with runtime.lazy_load_local_models. "
```

Comment:
But a test case for the same would still be helpful!

Reply:
The original error statement was handling that use case - 
"Must set runtime.local_models_dir with runtime.lazy_load_local_models" -> we just need to make sure that we include that too. So it needs to both be set AND be valid if `lazy_load_local_models` is True

Reply:
I can totally see 2 different error conditions for the 2 scenarios? 🤔 Let me know what you think

Reply:
In my test cases, when I do not set `local_models_dir` it is defaulting to `models` in the logic. 
So its picking it up from [somewhere](https://github.com/caikit/caikit/blob/6677470dd6b1896a7705e744277ae1fd5bf3a990/caikit/config/config.yml#L95)  

Which makes it difficult to assume, if the user did not set it or its default. I can delete that entry and then have logic, that checks if `local_models_dir` was not set. 
Is that the scenario we want, then I can make that change. But it will affect the config in the code base which does not make sense. 
Is there an alternative config I can add to the repo to load as part of the test?

Reply:
Hmm, the test config should be isolated from the base config and it should not be defaulting to `models`. Try to see where that's happening?